frontend/Store.js

```diff
@@ -141,6 +143,24 @@ class Store extends EventEmitter {
     });
 
     this._establishConnection();
+    this._eventQueue = [];
+    this._eventTimer = null;
+  }
+
+  emit(evt: string): boolean {
+    if (!this._eventTimer) {
+      this._eventTimer = setTimeout(() => {
+        this._eventQueue.forEach(evt => EventEmitter.prototype.emit.call(this, evt));
+        this._eventQueue = [];
+        this._eventTimer = null;
+      }, 50);
+      this._eventQueue = [];
+    }
+    if (this._eventQueue.indexOf(evt) === -1) {
+      this._eventQueue.push(evt);
+    }
+    // to appease flow
```

Comment:
imo it "should" return void. But flow wants a bool to align with the signature of EventEmitter. The bool indicates whether the emit "succeeded"